lib/pc_patch.c

```diff
@@ -724,3 +724,83 @@ pc_patch_set_schema(PCPATCH *patch, const PCSCHEMA *new_schema, double def)
 
 	return (PCPATCH*) paout;
 }
+
+
+/**
+* Read all the points from "patch", and transform them based on "new_schema".
+* Return a new patch with the transformed points.
+*/
+PCPATCH*
+pc_patch_transform(const PCPATCH *patch, const PCSCHEMA *new_schema, double def)
+{
+	PCDIMENSION** new_dimensions = new_schema->dims;
+	PCDIMENSION* old_dimensions[new_schema->ndims];
+	const PCSCHEMA *old_schema = patch->schema;
+	PCPATCH_UNCOMPRESSED *paout;
+	PCPOINT opt, npt;
+	PCPATCH *pain;
+	size_t i, j;
+
+	if ( old_schema->srid != new_schema->srid )
+	{
+		pcwarn("old and new schemas have different srids, and data "
+			   "reprojection is not yet supported");
+		return NULL;
+	}
+
+	for ( j = 0; j < new_schema->ndims; j++ )
+	{
+		PCDIMENSION *ndim = new_dimensions[j];
+		PCDIMENSION *odim = pc_schema_get_dimension_by_name(
+				old_schema, ndim->name);
+		old_dimensions[j] = odim;
+	}
+
```

Comment:
> But, as you said, if we consider this case as a normal condition (no warning message), then we should switch to a faster path instead of looping over each points for each dimensions into the new schema.

We still need a loop to create points, and an inner loop to create dimension values for each point. So the current code doesn't sound that bad for this case in my opinion. How would you do it differently?